bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -951,17 +1387,20 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
                 builder.parseLayerData(layer, mapLayerJson, this);
             }
 
-            if(mapLayerJson.created && isNaN(Date.parse(mapLayerJson.created)) === false){
+            if (mapLayerJson.created && isNaN(Date.parse(mapLayerJson.created)) === false) {
                 var created = new Date(mapLayerJson.created);
-                if(created) {
+                if (created) {
                     layer.setCreated(created);
                 }
             }
 
+            layer.setGroups(mapLayerJson.groups);
```

Comment:
Looks like the server response you are expecting is to have groups that have layers that have groups. Since layers need much more information than groups I would suggest that you review the datamodel that's being sent from the server. Maybe have layers in one array, groups in another and just use id's to link between them on the client-side.